receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -48,22 +53,58 @@ func newReceiver(params receiver.CreateSettings, config *Config, consumer consum
 		return nil, err
 	}
 
-	return &k8sobjectsreceiver{
-		client:   client,
-		setting:  params,
-		consumer: consumer,
-		objects:  config.Objects,
-		obsrecv:  obsrecv,
-		mu:       sync.Mutex{},
-	}, nil
+	objReceiver := &k8sobjectsreceiver{
+		client:         client,
+		setting:        params,
+		logger:         params.Logger,
+		consumer:       consumer,
+		objects:        config.Objects,
+		obsrecv:        obsrecv,
+		mu:             sync.Mutex{},
+		leaderElection: config.LeaderElection,
```

Comment:
Should this be prefixed with the component's name? Or is it OK to have the same leader ID for two components on the same instance?

Reply:
nice catch, I will add the component's name as electionId's prefix.